scholia/tex.py

```diff
@@ -252,13 +252,17 @@ def extract_qs_from_aux_string(string):
     >>> extract_qs_from_aux_string(string)
     ['Q28042913', 'Q27615040']
 
+    >>> string = "\citation{Q28042913,Q27615040.Q27615040}"
+    >>> extract_qs_from_aux_string(string)
+    ['Q28042913']
+
     """
     matches = re.findall(r'^\\citation{(.+?)}', string,
                          flags=re.MULTILINE | re.UNICODE)
     qs = []
     for submatches in matches:
         for q in submatches.split(','):
-            if re.match(r'Q\d+', q):
+            if re.match(r'^Q\d+$', q):
```

Comment:
If this query gets run over and over, it might be good to pre-compile the regular expression as a constant in the module, then later apply it like:

```python
wd_re = re.compile(r'^Q\d+$')
...
if wd_re.match(q):
    ...

```

Reply:
Good point.

Reply:
Yes, I suppose. For now, I have just merged it. 

If we want to make it more efficient, I think that the query to Wikidata can be made more efficient.